iris_grib/_save_rules.py

```diff
@@ -731,13 +733,33 @@ def set_forecast_time(cube, grib):
     gribapi.grib_set(grib, "forecastTime", fp)
 
 
-def set_fixed_surfaces(cube, grib):
+def set_fixed_surfaces(cube, grib, full3d_cube=None):
 
     # Look for something we can export
     v_coord = grib_v_code = output_unit = None
 
+    # checkout factories for hybrid vertical coordinates.
+    hybrid_height_factories = [
+        factory for factory in cube.aux_factories
+        if isinstance(factory, HybridHeightFactory)]
+    if not hybrid_height_factories:
+        hybrid_height_factory = None
+    else:
+        # N.B. get the matching 'complete' factory from the full 3d cube, so we
+        # have all the level information.
+        factory, = hybrid_height_factories
```

Comment:
Your implementation above seems to suggest that we could have a list of HybridHeightFactories (although I would consider it unlikely...).  If that happens, then the unpacked variable `factory` would become a list and then not have a name.

Would this be resolved by saying:
`factory, = hybrid_height_factories[0]`?

Reply:
> seems to suggest that we could have a list of HybridHeightFactories

That wasn't intended.
In fact, it's just a sneaky way of fetching the first of a list while simultaneously forcing an error if it doesn't have exactly one element.
It's handy, but every time I write one, I wonder if it's just a bit too clever ...

Reply:
Sorted.  Not a problem.